code/modules/antagonists/traitor/objectives/bug_room.dm

```diff
@@ -0,0 +1,187 @@
+/datum/traitor_objective_category/bug_room
+	name = "Bug Room"
+	objectives = list(
+		/datum/traitor_objective/bug_room = 1,
+		/datum/traitor_objective/bug_room/risky = 1,
+		/datum/traitor_objective/bug_room/super_risky = 1,
+	)
+
+/datum/traitor_objective/bug_room
+	name = "Bug the \[DEPARTMENT HEAD]'s office"
+	description = "Use the button below to materialize the bug within your hand, where you'll then be able to place it down in the \[DEPARTMENT HEAD]'s office. If it gets destroyed before you are able to plant it, this objective will fail."
+
+	progression_reward = list(2 MINUTES, 8 MINUTES)
+	telecrystal_reward = list(0, 1)
+
+	var/list/applicable_heads = list(
+		"Research Director" = /area/command/heads_quarters/rd,
+		"Chief Medical Officer" = /area/command/heads_quarters/cmo,
+		"Chief Engineer" = /area/command/heads_quarters/ce,
+		"Head of Personnel" = /area/command/heads_quarters/hop,
+		"Captain" = /area/command/heads_quarters/captain, // For head roles so that they can still get this objective.
+	)
+	var/datum/job/target_office
+	var/requires_head_as_supervisor = TRUE
+
+	var/obj/item/traitor_bug/bug
+
+/datum/traitor_objective/bug_room/risky
+	progression_minimum = 10 MINUTES
+	applicable_heads = list(
+		"Captain" = /area/command/heads_quarters/captain,
+	)
+	progression_reward = list(5 MINUTES, 10 MINUTES)
+	telecrystal_reward = list(1, 2)
+	requires_head_as_supervisor = FALSE
+
+/datum/traitor_objective/bug_room/super_risky
+	progression_minimum = 20 MINUTES
+	applicable_heads = list(
+		"Head of Security" = /area/command/heads_quarters/hos,
+	)
+	progression_reward = list(10 MINUTES, 15 MINUTES)
+	telecrystal_reward = list(2, 3)
+	requires_head_as_supervisor = FALSE
+
+/datum/traitor_objective/bug_room/super_risky/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	if(!handler.get_completion_count(/datum/traitor_objective/bug_room/risky))
+		// Locked if they don't have any of the risky bug room objective completed
+		return FALSE
+	return ..()
+
+/datum/traitor_objective/bug_room/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if(!bug)
+		buttons += add_ui_button("", "Pressing this will materialize a bug in your hand, which you can place at the target office", "wifi", "summon_gear")
+	return buttons
+
+/datum/traitor_objective/bug_room/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if("summon_gear")
+			if(bug)
+				return
+			bug = new(user.drop_location())
+			user.put_in_hands(bug)
```

Comment:
We should only let them do it if it can be placed in their hands, since I can't imagine a case where they wouldn't want that (that we'd be supporting)

Reply:
It'll just autodrop to the floor if both hands are full